.github/workflows/cpp.yml

```diff
@@ -0,0 +1,89 @@
+name: ci
+
+on:
+  workflow_call:
+    inputs:
+      style-branch:
+        type: string
+        default: master
+      continue-on-error:
+        type: boolean
+        default: false
+      files:
+        type: string
+    outputs:
+      passed-clang-format:
+        value: ${{ jobs.clang-format.outputs.passed }}
+      passed-roslint:
+        value: ${{ jobs.roslint.outputs.passed }}
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    outputs:
+      passed: ${{ steps.check.outputs.passed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Remove ignored paths
+        run: find . -name .lint-ignore | xargs dirname | xargs rm -rf
+      - name: Install tools
+        run: |
+          sudo apt-get install -y clang-format-11
+          curl -LOJ https://raw.githubusercontent.com/seqsense/ros_style/${{ inputs.style-branch }}/.clang-format
+      - name: Format
+        id: check
+        continue-on-error: ${{ inputs.continue-on-error }}
+        run: |
+          set -eo pipefail
+
+          files_to_check="${{ inputs.files }}"
+          if [ -z ${files_to_check} ]; then
+            files_to_check=$(find . -name "*.cpp" -o -name "*.h")
+          fi
+
+          (
+            set -eo pipefail
+            clang-format \
+                --style=file --dry-run --Werror \
+                ${files_to_check} \
+                2>&1 > /dev/null \
+              | sed ':l; N; /\n\S\+:[0-9]\+:[0-9]\+: error: /b l; N; s/\n/%0A/g' \
+              | sed 's/^\([^:]\+\):\([0-9]\+\):\([0-9]\+\): error: \+\(.*\)/::error file=\1,line=\2,col=\3::\4/'
+          ) && echo "::set-output name=passed::true" \
+            || (echo "::set-output name=passed::false"; exit 1)
+  roslint:
+    runs-on: ubuntu-latest
+    outputs:
+      passed: ${{ steps.check.outputs.passed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Remove ignored paths
+        run: find . -name .lint-ignore | xargs dirname | xargs rm -rf
+      - name: Install tools
+        run: sudo pip3 install --upgrade git+https://github.com/at-wat/roslint-pip
+      - name: Lint
+        id: check
+        continue-on-error: ${{ inputs.continue-on-error }}
+        run: |
+          set -eo pipefail
+
+          files_to_check="${{ inputs.files }}"
+          if [ -z ${files_to_check} ]; then
+            files_to_check=$(find . -name "*.cpp" -o -name "*.h")
+          fi
+
+          (
+            set -eo pipefail
+            roslint-cpplint \
+                --filter=-runtime/references,-build/c++11 \
```

Comment:
Maybe the filter could be based on an argument or loaded from the file and default to `-runtime/references,-build/c++11`.